### DISCUSSION TITLE
PostgreSql schema

### DISCUSSION THREAD

**Discussion by lazyclaw-labs** [linq2db/linq2db#3449] [General]
I use PostgreSql dataprovider.getschema, and got back tables schemas. But I have questions on values of the schema:

1. TableSchema: Type property. Isn't this supposed to be the class name? I found it to be table name instead.
2. ColumnSchema: MemberName, isn't this supposed to be the property name? I found it to be the db columnname instead. There's a column name on the class too. MemberName is always the same as Column name, it doesn't make sense to me.

**Comment by MaceWindu** [comment]
Well, this API was designed for T4 templates and a lot of stuff in it doesn't make much sense when used directly - it contains a lot of T4 specific information like properties you mentioned to help with code generation by T4 template.

Currently we are working on replacement for T4 templates (expected to release initial preview with next v4 preview release) and later plan to introduce new schema read API, which will contain only schema information.

**Reply by lazyclaw-labs** [reply]
Got it. Have to say it's a bad decision. We should build t4 around schema, but schema around t4. We use schema information quite seriously, got this unexpected problem. It worked fine with SQL server. :-)

**Reply by MaceWindu** [reply]
Not sure I understand you. What is bad decision?
